presto-main/src/main/java/com/facebook/presto/operator/aggregation/AccumulatorFactory.java

```diff
@@ -30,4 +30,8 @@
     boolean hasOrderBy();
 
     boolean hasDistinct();
+
+    GroupedAccumulator createSpillableGroupedAccumulator();
+
+    GroupedAccumulator createSpillableGroupedIntermediateAccumulator();
```

Comment:
Instead of having these two new interfaces, we can make `spillingEnabled` a variable for `AccumulatorFactoryBinder`. `spillingEnabled` can be passed in through `AccumulatorFactoryBinder::bind`. In that way, we can avoid a lot of tunneling. `spillingEnabled` can be directly obtained from `LocalExecutionPlanner`

Reply:
Will update once completed offline discussion on this. May cause a bug.

Reply:
I'm not sure if checking if spill is enabled via `LocalExecutionPlanner` is correct. This is because of the way that HashAggOperator is constructed via its factory: see `createHashAggregationOperatorFactory` in `LocalExecutionPlanner`. It is explicitly created with `spillEnabled = false` in multiple places. 

This means that if we were to check if spill is enabled on bind it it will break this existing functionality so tunneling, while ugly, may be the only way to do this.

Reply:
Why is that? Isn't it created from `planGroupByAggregation` with `spillEnabled` as a variable?